pixeltable/tests/test_exprs.py

```diff
@@ -356,6 +358,44 @@ def test_arrays(self, test_tbl: catalog.Table) -> None:
         _ = t[t.array_col[:, 0]].show()
         print(_)
 
+    def test_astype(self, test_tbl: catalog.Table) -> None:
+        t = test_tbl
+        # Convert int to float
+        status = t.add_column(c2_as_float=t.c2.astype(FloatType()))
+        assert status.num_excs == 0
+        data = t.select(t.c2, t.c2_as_float).collect()
+        for row in data:
+            assert isinstance(row['c2'], int)
+            assert isinstance(row['c2_as_float'], float)
+            assert row['c2'] == row['c2_as_float']
```

Comment:
This is missing failure scenarios (with pytest.raises()...)

Also, test astype() on something other than a ColumnRef.